compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -327,9 +327,52 @@ object Trees {
 
   extension (mdef: untpd.DefTree) def mods: untpd.Modifiers = mdef.rawMods
 
-  abstract class NamedDefTree[-T >: Untyped](implicit @constructorOnly src: SourceFile) extends NameTree[T] with DefTree[T] {
+  /** PackageDef | NamedDefTree */
```

Comment:
I assume that comment is supposed to mean that only PackageDef or NameDefTree extend that trait? If so that could be written explicitly (or perhaps the self-type could be `PackageDef | NamedDefTree` ?). Also the documentation comment should explain what this trait is for.

Reply:
Originally its because adding the type parameter made it more complex - but I've adjusted now to using the self type